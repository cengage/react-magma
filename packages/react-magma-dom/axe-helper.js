import { configureAxe } from 'jest-axe';

export const axe = configureAxe({
  rules: {
    // disabled landmark rules when testing isolated components.
    region: { enabled: false },
    // disabled nested-interactive when testing isolated components
    'nested-interactive': { enabled: false },
  },
});
