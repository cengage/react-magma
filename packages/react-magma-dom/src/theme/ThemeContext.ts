import * as React from 'react';

import { magma } from './magma';

export const ThemeContext = React.createContext(
  magma // default value
);
