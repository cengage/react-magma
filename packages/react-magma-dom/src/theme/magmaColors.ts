export enum MagmaColors {
  primary = 'primary',
  secondary = 'secondary',
  tertiary = 'tertiary',
  neutral = 'neutral',
  info = 'info',
  danger = 'danger',
  warning = 'warning',
  success = 'success',
}
