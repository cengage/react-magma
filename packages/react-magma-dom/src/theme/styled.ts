import styled from '@emotion/styled-base';
import { CreateStyled } from '@emotion/styled/types';
import { ThemeInterface } from './magma';

export default styled as CreateStyled<ThemeInterface>;
