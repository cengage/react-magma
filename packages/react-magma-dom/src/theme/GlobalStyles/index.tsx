import * as React from 'react';
import { Global, css } from '@emotion/core';
import { ThemeContext } from '../ThemeContext';
import { useIsInverse } from '../../inverse';

function getStyles(theme, isInverse: boolean) {
  return css`
    @import url('https://fonts.googleapis.com/css2?family=Work+Sans:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap');
    @import url('https://fonts.googleapis.com/css2?family=Noto+Serif:ital,wght@0,400;0,700;1,400;1,700&display=swap');

    *,
    *:before,
    *:after {
      box-sizing: border-box;
    }

    *:focus {
      outline: 2px solid
        ${isInverse ? theme.colors.focusInverse : theme.colors.focus};
      outline-offset: 2px;
    }

    html {
      -ms-text-size-adjust: 100%;
      -webkit-text-size-adjust: 100%;
      font-size: ${theme.typeScale.size03.fontSize};
      line-height: ${theme.typeScale.size03.lineHeight};
      scroll-behavior: smooth;
    }

    html,
    body {
      background: ${isInverse
        ? theme.colors.foundation
        : theme.colors.neutral08};
      color: ${isInverse ? theme.colors.neutral08 : theme.colors.neutral};
      margin: 0;
      padding: 0;
    }

    body {
      font-family: ${theme.bodyFont};
      font-style: normal;
      font-weight: 400;
      font-size: ${theme.typeScale.size03.fontSize};
      line-height: ${theme.typeScale.size03.lineHeight};
    }

    a {
      color: ${isInverse ? theme.colors.neutral07 : theme.colors.primary};
      cursor: pointer;
      text-decoration: underline;

      &:hover,
      &:focus {
        color: ${isInverse ? theme.colors.neutral06 : theme.colors.focus};
      }
    }

    button,
    input,
    optgroup,
    select,
    textarea {
      font: inherit;
      margin: 0;
    }

    svg:not(:root) {
      overflow: hidden;
    }
  `;
}

export const GlobalStyles: React.FunctionComponent = () => {
  const isInverse = useIsInverse();
  return (
    <ThemeContext.Consumer>
      {theme => <Global styles={getStyles(theme, isInverse)} />}
    </ThemeContext.Consumer>
  );
};
