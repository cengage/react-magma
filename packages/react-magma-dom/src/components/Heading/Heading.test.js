import React from 'react';
import { axe } from '../../../axe-helper';
import { Heading } from '.';
import { render } from '@testing-library/react';
import { magma } from '../../theme/magma';
import { TypographyContextVariant, TypographyVisualStyle } from '../Typography';

describe('Heading', () => {
  it('should find element by testId', () => {
    const testId = 'test-id';
    const { getByTestId } = render(
      <Heading level={1} testId={testId}>
        test
      </Heading>
    );

    expect(getByTestId(testId)).toBeInTheDocument();
  });

  it('should render an h1', () => {
    const headingText = 'test';
    const { getByText } = render(<Heading level={1}>{headingText}</Heading>);
    const heading = getByText(headingText);

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(headingText);
    expect(heading).toHaveStyleRule(
      'font-size',
      magma.typeScale.size07.fontSize
    );
    expect(heading).toHaveStyleRule(
      'margin',
      `0 0 ${magma.spaceScale.spacing05}`
    );
  });

  it('should render an h2', () => {
    const headingText = 'test';
    const { getByText } = render(<Heading level={2}>{headingText}</Heading>);
    const heading = getByText(headingText);

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(headingText);
    expect(heading).toHaveStyleRule(
      'font-size',
      magma.typeScale.size06.fontSize
    );
    expect(heading).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing10} 0 ${magma.spaceScale.spacing05}`
    );
  });

  it('should render an h3', () => {
    const headingText = 'test';
    const { getByText } = render(<Heading level={3}>{headingText}</Heading>);
    const heading = getByText(headingText);

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(headingText);
    expect(heading).toHaveStyleRule(
      'font-size',
      magma.typeScale.size05.fontSize
    );
    expect(heading).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing09} 0 ${magma.spaceScale.spacing05}`
    );
  });

  it('should render an h4', () => {
    const headingText = 'test';
    const { getByText } = render(<Heading level={4}>{headingText}</Heading>);
    const heading = getByText(headingText);

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(headingText);
    expect(heading).toHaveStyleRule(
      'font-size',
      magma.typeScale.size04.fontSize
    );
    expect(heading).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing08} 0 ${magma.spaceScale.spacing05}`
    );
  });

  it('should render an h5', () => {
    const headingText = 'test';
    const { getByText } = render(<Heading level={5}>{headingText}</Heading>);
    const heading = getByText(headingText);

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(headingText);
    expect(heading).toHaveStyleRule(
      'font-size',
      magma.typeScale.size04.fontSize
    );
    expect(heading).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing06} 0 ${magma.spaceScale.spacing05}`
    );
  });

  it('should render an h6', () => {
    const headingText = 'test';
    const { getByText } = render(<Heading level={6}>{headingText}</Heading>);
    const heading = getByText(headingText);

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(headingText);
    expect(heading).toHaveStyleRule(
      'font-size',
      magma.typeScale.size01.fontSize
    );
    expect(heading).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing06} 0 ${magma.spaceScale.spacing03}`
    );
  });

  it('should render an component with a different size', () => {
    const headingText = 'test';
    const { container, getByText } = render(
      <Heading visualStyle="headingXLarge" level={6}>
        {headingText}
      </Heading>
    );
    const heading = getByText(headingText);

    expect(container.querySelector('h6')).toBeInTheDocument();
    expect(heading).toHaveTextContent(headingText);
    expect(heading).toHaveStyleRule(
      'font-size',
      magma.typeScale.size07.fontSize
    );
  });

  it('should render headings without margins styles', () => {
    const headingText1Exp = 'test 1 expressive';
    const headingText1 = 'test 1';
    const headingText2 = 'test 2';
    const headingText3 = 'test 3';
    const headingText4 = 'test 4';
    const headingText5 = 'test 5';
    const headingText6 = 'test 6';

    const { getByText } = render(
      <>
        <Heading
          noMargins
          level={1}
          contextVariant={TypographyContextVariant.expressive}
          visualStyle={TypographyVisualStyle.heading2XLarge}
        >
          {headingText1Exp}
        </Heading>
        <Heading noMargins level={1}>
          {headingText1}
        </Heading>
        <Heading noMargins level={2}>
          {headingText2}
        </Heading>
        <Heading noMargins level={3}>
          {headingText3}
        </Heading>
        <Heading noMargins level={4}>
          {headingText4}
        </Heading>
        <Heading noMargins level={5}>
          {headingText5}
        </Heading>
        <Heading noMargins level={6}>
          {headingText6}
        </Heading>
      </>
    );

    expect(getByText(headingText1Exp)).toHaveStyleRule('margin', '0');
    expect(getByText(headingText1)).toHaveStyleRule('margin', '0');
    expect(getByText(headingText2)).toHaveStyleRule('margin', '0');
    expect(getByText(headingText3)).toHaveStyleRule('margin', '0');
    expect(getByText(headingText4)).toHaveStyleRule('margin', '0');
    expect(getByText(headingText5)).toHaveStyleRule('margin', '0');
    expect(getByText(headingText6)).toHaveStyleRule('margin', '0');
  });

  it('should render headings without top margin styles', () => {
    const headingText1Exp = 'test 1 expressive';
    const headingText1 = 'test 1';
    const headingText2 = 'test 2';
    const headingText3 = 'test 3';
    const headingText4 = 'test 4';
    const headingText5 = 'test 5';
    const headingText6 = 'test 6';

    const { getByText } = render(
      <>
        <Heading
          noTopMargin
          level={1}
          contextVariant={TypographyContextVariant.expressive}
          visualStyle={TypographyVisualStyle.heading2XLarge}
        >
          {headingText1Exp}
        </Heading>
        <Heading noTopMargin level={1}>
          {headingText1}
        </Heading>
        <Heading noTopMargin level={2}>
          {headingText2}
        </Heading>
        <Heading noTopMargin level={3}>
          {headingText3}
        </Heading>
        <Heading noTopMargin level={4}>
          {headingText4}
        </Heading>
        <Heading noTopMargin level={5}>
          {headingText5}
        </Heading>
        <Heading noTopMargin level={6}>
          {headingText6}
        </Heading>
      </>
    );

    expect(getByText(headingText1Exp)).toHaveStyleRule(
      'margin',
      `0 0 ${magma.spaceScale.spacing06} 0`
    );
    expect(getByText(headingText1)).toHaveStyleRule(
      'margin',
      `0 0 ${magma.spaceScale.spacing05} 0`
    );
    expect(getByText(headingText2)).toHaveStyleRule(
      'margin',
      `0 0 ${magma.spaceScale.spacing05} 0`
    );
    expect(getByText(headingText3)).toHaveStyleRule(
      'margin',
      `0 0 ${magma.spaceScale.spacing05} 0`
    );
    expect(getByText(headingText4)).toHaveStyleRule(
      'margin',
      `0 0 ${magma.spaceScale.spacing05} 0`
    );
    expect(getByText(headingText5)).toHaveStyleRule(
      'margin',
      `0 0 ${magma.spaceScale.spacing05} 0`
    );
    expect(getByText(headingText6)).toHaveStyleRule(
      'margin',
      `0 0 ${magma.spaceScale.spacing03} 0`
    );
  });

  it('should render headings without bottom margin styles', () => {
    const headingText1Exp = 'test 1 expressive';
    const headingText1 = 'test 1';
    const headingText2 = 'test 2';
    const headingText3 = 'test 3';
    const headingText4 = 'test 4';
    const headingText5 = 'test 5';
    const headingText6 = 'test 6';

    const { getByText } = render(
      <>
        <Heading
          noBottomMargin
          level={1}
          contextVariant={TypographyContextVariant.expressive}
          visualStyle={TypographyVisualStyle.heading2XLarge}
        >
          {headingText1Exp}
        </Heading>
        <Heading noBottomMargin level={1}>
          {headingText1}
        </Heading>
        <Heading noBottomMargin level={2}>
          {headingText2}
        </Heading>
        <Heading noBottomMargin level={3}>
          {headingText3}
        </Heading>
        <Heading noBottomMargin level={4}>
          {headingText4}
        </Heading>
        <Heading noBottomMargin level={5}>
          {headingText5}
        </Heading>
        <Heading noBottomMargin level={6}>
          {headingText6}
        </Heading>
      </>
    );

    expect(getByText(headingText1Exp)).toHaveStyleRule('margin', '0');
    expect(getByText(headingText1)).toHaveStyleRule('margin', `0`);
    expect(getByText(headingText2)).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing10} 0 0 0`
    );
    expect(getByText(headingText3)).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing09} 0 0 0`
    );
    expect(getByText(headingText4)).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing08} 0 0 0`
    );
    expect(getByText(headingText5)).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing06} 0 0 0`
    );
    expect(getByText(headingText6)).toHaveStyleRule(
      'margin',
      `${magma.spaceScale.spacing06} 0 0 0`
    );
  });

  it('should render inverse styles', () => {
    const headingText = 'test';
    const { getByText } = render(
      <Heading isInverse level={1}>
        {headingText}
      </Heading>
    );
    const heading = getByText(headingText);

    expect(heading).toHaveStyleRule('color', magma.colors.neutral100);

    expect(heading).toHaveStyleRule(
      'border-bottom',
      `2px solid ${magma.colors.focusInverse}`,
      {
        target: ':focus',
      }
    );
  });

  it('should render custom styles', () => {
    const color = '#cccccc';
    const headingText = 'test';
    const { getByText } = render(
      <Heading level={1} style={{ color }}>
        {headingText}
      </Heading>
    );
    const heading = getByText(headingText);

    expect(heading).toHaveStyle(`color: ${color}`);
  });

  describe('Snapshot', () => {
    it('should render heading 1 correctly', () => {
      const { container } = render(
        <Heading level={1} id="testId">
          Heading 1
        </Heading>
      );

      expect(container).toMatchSnapshot();
    });

    it('should render heading 2 correctly', () => {
      const { container } = render(<Heading level={2}>Heading 2</Heading>);

      expect(container).toMatchSnapshot();
    });

    it('should render heading 3 correctly', () => {
      const { container } = render(<Heading level={3}>Heading 3</Heading>);

      expect(container).toMatchSnapshot();
    });

    it('should render heading 4 correctly', () => {
      const { container } = render(<Heading level={4}>Heading 4</Heading>);

      expect(container).toMatchSnapshot();
    });

    it('should render heading 5 correctly', () => {
      const { container } = render(<Heading level={5}>Heading 5</Heading>);

      expect(container).toMatchSnapshot();
    });

    it('should render heading 6 correctly', () => {
      const { container } = render(<Heading level={6}>Heading 6</Heading>);

      expect(container).toMatchSnapshot();
    });

    it('should render expressive heading2XLarge correctly', () => {
      const { container } = render(
        <Heading
          level={1}
          contextVariant={TypographyContextVariant.expressive}
          visualStyle={TypographyVisualStyle.heading2XLarge}
          id="testId"
        >
          Heading 1
        </Heading>
      );

      expect(container).toMatchSnapshot();
    });

    it('should render expressive headingXLarge correctly', () => {
      const { container } = render(
        <Heading
          level={2}
          contextVariant={TypographyContextVariant.expressive}
          visualStyle={TypographyVisualStyle.headingXLarge}
          id="testId"
        >
          Heading 2
        </Heading>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('expressive styles', () => {
    it('should render expressive h1 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="expressive" level={1}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule('color', magma.colors.primary600);
      expect(heading).toHaveStyleRule(
        'font-family',
        magma.headingExpressiveFont
      );
      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyExpressiveVisualStyles.headingXLarge.mobile.fontSize
      );
    });

    it('should render expressive h2 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="expressive" level={2}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyExpressiveVisualStyles.headingLarge.mobile.fontSize
      );
    });

    it('should render expressive h3 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="expressive" level={3}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyExpressiveVisualStyles.headingMedium.mobile.fontSize
      );
    });

    it('should render expressive h4 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="expressive" level={4}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyExpressiveVisualStyles.headingSmall.mobile.fontSize
      );
    });

    it('should render expressive h5 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="expressive" level={5}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyExpressiveVisualStyles.headingXSmall.mobile.fontSize
      );
    });

    it('should render expressive h6 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="expressive" level={6}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyExpressiveVisualStyles.heading2XSmall.mobile.fontSize
      );
    });
  });

  describe('narrative styles', () => {
    it('should render narrative h1 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="narrative" level={1}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule('color', magma.colors.neutral700);
      expect(heading).toHaveStyleRule(
        'font-family',
        magma.headingNarrativeFont
      );
      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyNarrativeVisualStyles.headingXLarge.mobile.fontSize
      );
    });

    it('should render narrative h2 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="narrative" level={2}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyNarrativeVisualStyles.headingLarge.mobile.fontSize
      );
    });

    it('should render narrative h3 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="narrative" level={3}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyNarrativeVisualStyles.headingMedium.mobile.fontSize
      );
    });

    it('should render narrative h4 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="narrative" level={4}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyNarrativeVisualStyles.headingSmall.mobile.fontSize
      );
    });

    it('should render narrative h5 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="narrative" level={5}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyNarrativeVisualStyles.headingXSmall.mobile.fontSize
      );
    });

    it('should render narrative h6 styles', () => {
      const headingText = 'test';
      const { getByText } = render(
        <Heading contextVariant="narrative" level={6}>
          {headingText}
        </Heading>
      );
      const heading = getByText(headingText);

      expect(heading).toHaveStyleRule(
        'font-size',
        magma.typographyNarrativeVisualStyles.heading2XSmall.mobile.fontSize
      );
    });
  });

  it('Does not violate accessibility standards', () => {
    const { container } = render(<Heading level={1}>test</Heading>);
    return axe(container.innerHTML).then(result => {
      return expect(result).toHaveNoViolations();
    });
  });
});
