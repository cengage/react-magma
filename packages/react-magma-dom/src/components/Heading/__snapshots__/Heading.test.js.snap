// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Snapshot should render expressive heading2XLarge correctly 1`] = `
.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #292F7C;
  font-size: 48px;
  font-weight: 500;
  line-height: 64px;
  margin: 0 0 24px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 64px;
    line-height: 84px;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #292F7C;
  font-size: 48px;
  font-weight: 500;
  line-height: 64px;
  margin: 0 0 24px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 64px;
    line-height: 84px;
  }
}

<div>
  <h1
    class="emotion-0 emotion-1"
    id="testId"
  >
    Heading 1
  </h1>
</div>
`;

exports[`Heading Snapshot should render expressive headingXLarge correctly 1`] = `
.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #292F7C;
  font-size: 28px;
  font-weight: 600;
  line-height: 40px;
  margin: 0 0 16px;
  font-size: 36px;
  font-weight: 600;
  line-height: 48px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 36px;
    line-height: 48px;
  }
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 48px;
    line-height: 64px;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #292F7C;
  font-size: 28px;
  font-weight: 600;
  line-height: 40px;
  margin: 0 0 16px;
  font-size: 36px;
  font-weight: 600;
  line-height: 48px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 36px;
    line-height: 48px;
  }
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 48px;
    line-height: 64px;
  }
}

<div>
  <h2
    class="emotion-0 emotion-1"
    id="testId"
  >
    Heading 2
  </h2>
</div>
`;

exports[`Heading Snapshot should render heading 1 correctly 1`] = `
.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 28px;
  font-weight: 600;
  line-height: 40px;
  margin: 0 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 36px;
    line-height: 48px;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 28px;
  font-weight: 600;
  line-height: 40px;
  margin: 0 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 36px;
    line-height: 48px;
  }
}

<div>
  <h1
    class="emotion-0 emotion-1"
    id="testId"
  >
    Heading 1
  </h1>
</div>
`;

exports[`Heading Snapshot should render heading 2 correctly 1`] = `
.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 24px;
  font-weight: 600;
  line-height: 32px;
  margin: 48px 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 28px;
    line-height: 40px;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 24px;
  font-weight: 600;
  line-height: 32px;
  margin: 48px 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 28px;
    line-height: 40px;
  }
}

<div>
  <h2
    class="emotion-0 emotion-1"
  >
    Heading 2
  </h2>
</div>
`;

exports[`Heading Snapshot should render heading 3 correctly 1`] = `
.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 20px;
  font-weight: 600;
  line-height: 32px;
  margin: 40px 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 24px;
    line-height: 32px;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 20px;
  font-weight: 600;
  line-height: 32px;
  margin: 40px 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 24px;
    line-height: 32px;
  }
}

<div>
  <h3
    class="emotion-0 emotion-1"
  >
    Heading 3
  </h3>
</div>
`;

exports[`Heading Snapshot should render heading 4 correctly 1`] = `
.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 18px;
  font-weight: 600;
  line-height: 24px;
  margin: 32px 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 20px;
    line-height: 32px;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 18px;
  font-weight: 600;
  line-height: 24px;
  margin: 32px 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 20px;
    line-height: 32px;
  }
}

<div>
  <h4
    class="emotion-0 emotion-1"
  >
    Heading 4
  </h4>
</div>
`;

exports[`Heading Snapshot should render heading 5 correctly 1`] = `
.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 18px;
  font-weight: 600;
  line-height: 24px;
  margin: 24px 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 18px;
    line-height: 24px;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 18px;
  font-weight: 600;
  line-height: 24px;
  margin: 24px 0 16px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 18px;
    line-height: 24px;
  }
}

<div>
  <h5
    class="emotion-0 emotion-1"
  >
    Heading 5
  </h5>
</div>
`;

exports[`Heading Snapshot should render heading 6 correctly 1`] = `
.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 12px;
  font-weight: 700;
  letter-spacing: .32px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 24px 0 8px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 12px;
    letter-spacing: .32px;
    line-height: 16px;
  }
}

.emotion-0 {
  border-bottom: 2px solid transparent;
  font-family: "Work Sans",Helvetica,sans-serif;
  padding: 0;
  color: #454545;
  font-size: 12px;
  font-weight: 700;
  letter-spacing: .32px;
  line-height: 16px;
  text-transform: uppercase;
  margin: 24px 0 8px;
}

.emotion-0:focus {
  border-bottom: 2px solid #0074B7;
  outline: 0;
  -webkit-transition: border 0.1s linear;
  transition: border 0.1s linear;
}

@media (min-width: 600px) {
  .emotion-0 {
    font-size: 12px;
    letter-spacing: .32px;
    line-height: 16px;
  }
}

<div>
  <h6
    class="emotion-0 emotion-1"
  >
    Heading 6
  </h6>
</div>
`;
